src/simtools/production_configuration/limits_calculation.py

```diff
@@ -68,44 +115,49 @@ def _compute_limits(
         float
             Bin edge value corresponding to the threshold.
         """
-        projection = np.sum(event_weight_array, axis=axis)
-        bin_edge_value = None
-        cumulative_sum = None
-        if limit_type == "upper":
-            cumulative_sum = np.cumsum(projection)
-
-        elif limit_type == "lower":
-            cumulative_sum = np.cumsum(projection[::-1])
-
+        projection = np.sum(hist, axis=axis)
+        cumulative_sum = (
+            np.cumsum(projection) if limit_type == "upper" else np.cumsum(projection[::-1])
+        )
         total_events = np.sum(projection)
         threshold = (1 - loss_fraction) * total_events
         bin_index = np.searchsorted(cumulative_sum, threshold)
-        if limit_type == "upper":
-            bin_edge_value = bin_edges[bin_index]
-        elif limit_type == "lower":
-            bin_edge_value = bin_edges[-bin_index]
+        bin_edge_value = bin_edges[bin_index] if limit_type == "upper" else bin_edges[-bin_index]
         return bin_index, bin_edge_value
 
-    def get_bin_edges_and_units(self, table, axis="x"):
+    def _prepare_data_for_limits(self):
         """
-        Extract bin edges and units from the table metadata.
-
-        Parameters
-        ----------
-        table : astropy.table.Table
-            Table containing the event data.
+        Prepare the data required for computing limits.
 
         Returns
         -------
         tuple
-            Tuple containing the bin edges and their units.
+            Tuple containing core distances, triggered energies, core bins, and energy bins.
         """
-        bin_edges = table.meta[f"{axis}_bin_edges"]
-        try:
-            bin_edges_unit = table.meta[f"{axis}_bin_edges_unit"]
-        except KeyError:
-            bin_edges_unit = ""
-        return bin_edges, bin_edges_unit
+        num_files = len(self.list_of_files)
+        showers_per_file = len(self.simulated) // num_files
```

Comment:
Can we have a zero division at this point? i think not, but please confirm. In my understanding we checked already earlier that there are files in the list.